Plugin returns "received message larger than max" error when sending payloads greater than 4194304 bytes

**Describe the bug**
When using a custom vault plugin and sending a payload greater than 4194304 bytes, the operation ends in failure due to the grpc default max receive message size of `1024 * 1024 * 4`. This option does not appear to be configurable, so servicing larger requests is not possible. 

**To Reproduce**
Steps to reproduce the behavior:
1. Write payload to plugin `vault write customplugin/action @large_28mb_payload`
2. See error: `[ERROR] core: failed to run existence check: error="rpc error: code = ResourceExhausted desc = grpc: received message larger than max (28793489 vs. 4194304)"`

**Expected behavior**
In `plugin.Server`, set the max receive size to a value similar to or the same as the max send size, which would align to the changes made for #4843.

Alternately, allow both the send and receive values to be configurable via `plugin.ServeOpts`.

**Environment:**
* Vault Server Version (retrieve with `vault status`): 0.10.3
* Vault CLI Version (retrieve with `vault version`): Vault v0.10.3 ('c69ae68faf2bf7fc1d78e3ec62655696a07454c7')
* Server Operating System/Architecture: darwin, linux

Vault server configuration file(s): Running in `-dev` mode.

**Additional context**
This impacts the ability to perform cryptographic operations (similar to that of transit engine) on larger binary files.

grpc declares these constant values as:
```
defaultServerMaxReceiveMessageSize = 1024 * 1024 * 4
defaultServerMaxSendMessageSize    = math.MaxInt32
```

--- a/logical/plugin/grpc_backend_client.go
+++ b/logical/plugin/grpc_backend_client.go
@@ -3,6 +3,7 @@ package plugin
 import (
 	"context"
 	"errors"
+	"math"
 	"sync/atomic"
 
 	"google.golang.org/grpc"
@@ -195,6 +196,9 @@ func (b *backendGRPCPluginClient) Setup(ctx context.Context, config *logical.Bac
 
 	// Register the server in this closure.
 	serverFunc := func(opts []grpc.ServerOption) *grpc.Server {
+		opts = append(opts, grpc.MaxRecvMsgSize(math.MaxInt32))
+		opts = append(opts, grpc.MaxSendMsgSize(math.MaxInt32))
+
 		s := grpc.NewServer(opts...)
 		pb.RegisterSystemViewServer(s, sysView)
 		pb.RegisterStorageServer(s, storage)

--- a/logical/plugin/serve.go
+++ b/logical/plugin/serve.go
@@ -2,7 +2,10 @@ package plugin
 
 import (
 	"crypto/tls"
+	"math"
 	"os"
+
+	"google.golang.org/grpc"
 
 	log "github.com/hashicorp/go-hclog"
 	"github.com/hashicorp/go-plugin"
@@ -54,7 +57,11 @@ func Serve(opts *ServeOpts) error {
 		Logger:          logger,
 
 		// A non-nil value here enables gRPC serving for this plugin...
-		GRPCServer: plugin.DefaultGRPCServer,
+		GRPCServer: func(opts []grpc.ServerOption) *grpc.Server {
+			opts = append(opts, grpc.MaxRecvMsgSize(math.MaxInt32))
+			opts = append(opts, grpc.MaxSendMsgSize(math.MaxInt32))
+			return plugin.DefaultGRPCServer(opts)
+		},
 	}
 
 	if !pluginutil.GRPCSupport() {

